### ISSUE TITLE
[a11y]: Toggle button has violation

### ISSUE THREAD

**Issue by aknguyen7** [carbon-design-system/carbon#18108]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.54.0

### React version

1.54.0

### Automated testing tool and ruleset

IBM Accessibility Checker version 7.3

### Description

We have the same violation when using the Toggle button in our code

### WCAG 2.1 Violation

Text contrast of 1.73 with its background is less than the WCAG AA minimum requirements for text of size 14px and weight of 400

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-toggle--playground&args=disabled:!true;size:sm;hideLabel:!true;toggled:!false;readOnly:!false;aria-labelledby:toggle-3

### Steps to reproduce

Using the playground from https://react.carbondesignsystem.com/?path=/story/components-toggle--playground&args=disabled:!true;size:sm;hideLabel:!true;toggled:!false;readOnly:!false;aria-labelledby:toggle-3 

I can recreate the problem with the following settings

![Image](https://github.com/user-attachments/assets/4ce98e72-3a93-4a60-a831-4e7e9f1d52ab)

![Image](https://github.com/user-attachments/assets/ad26ca0f-35aa-4060-97b3-ef49cd0c2384)

![Image](https://github.com/user-attachments/assets/fa8de528-fc36-46b5-b011-e472f8110552)

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
hey @aknguyen7 
The issue arises from the change you made to the `aria-labelledby` attribute in the `Playground`. When you updated it to` aria-labelledby="toggle3"`, it caused a mismatch because the id of the label is `toggle-3_label,` not `toggle3`
 The aria-labelledby attribute should reference the id of the label element (e.g., toggle-3_label) to ensure proper labeling for accessibility. This violation is not seen when aria-labelledby="toggle-3_label" [here](https://react.carbondesignsystem.com/iframe.html?args=aria-labelledby%3Atoggle-3_label%3Bdisabled%3A!true%3BhideLabel%3A!true%3BreadOnly%3A!false%3Bsize%3Asm%3Btoggled%3A!false&globals=&id=components-toggle--playground&viewMode=story)
